metricflow/dataflow/builder/source_node.py

```diff
@@ -95,3 +97,11 @@ def create_from_data_sets(self, data_sets: Sequence[SemanticModelDataSet]) -> So
             source_nodes_for_group_by_item_queries=tuple(group_by_item_source_nodes) + (self._time_spine_source_node,),
             source_nodes_for_metric_queries=tuple(source_nodes_for_metric_queries),
         )
+
+    def build_source_node_for_group_by_metric(self, group_by_metric_spec: GroupByMetricSpec) -> MetricFlowQuerySpec:
+        """Build source node used to satisfy requested group by metrics.
+
+        This is just a wrapper around the query parser method, stored here to limit the scope of the DataFlowPlanBuilder's
+        dependency on the query parser to only source nodes.
+        """
```

Comment:
Documentation and naming nit - this is not building the source node itself, it's actually providing the query spec so the DataflowPlanBuilder can do that. Maybe this instead?
```suggestion
    def build_source_node_inputs_for_group_by_metric(self, group_by_metric_spec: GroupByMetricSpec) -> MetricFlowQuerySpec:
        """Build source node inputs used to satisfy requested group by metrics.

        Group by metrics are essentially metric subqueries that operate as source nodes in
    the DataflowPlanBuilder. We provide the inputs here because they require an entire
    DataFlowPlan generation step.

    This is just a wrapper around the query parser method, stored here to limit the scope of the DataFlowPlanBuilder's
        dependency on the query parser to only source nodes.
        """
```